react/features/base/participants/reducer.ts

```diff
@@ -499,6 +500,33 @@ ReducerRegistry.register<IParticipantsState>('features/base/participants',
             raisedHandsQueue: action.queue
         };
     }
+    case UPDATE_CONFERENCE_METADATA: {
+        const { metadata } = action;
+
+
+        if (metadata?.visitors?.promoted) {
+            let participantProcessed = false;
+
+            Object.entries(metadata?.visitors?.promoted).forEach(([ key, _ ]) => {
+
+                const p = state.remote.get(key);
+
+                if (p && !p.isPromoted) {
+                    const newParticipant = _participant(p, action);
```

Comment:
As I'm reading this, the `_participant` function should return `p`. Isn't this true? 

You can either remove this line or add a line in the switch that can handle the `UPDATE_CONFERENCE_METADATA` action and update the isPromoted flag. IMHO the latter seems the direction we were trying to go.

NOTE: Also it will be good the new participant object that is added to be with a different reference. This way if some component or state listener is relying on this participant only (not the whole map) redux will trigger the update.

Reply:
Oh wait. You are actually almost doing this but the action doesn't have participant prop. this is why it doesn't update anything... Please fix this.